src/plugins/resourcelink.js

```diff
@@ -0,0 +1,58 @@
+let redact;
+
+const RESOURCELINK_RE = /^\[r ([^\]]+)\]/;
+const RESOURCELINK = 'resourcelink';
+
+/**
+ * Plugin that adds support for Curriculum Builder's resourcelinks.
```

Comment:
See https://github.com/remarkjs/remark/tree/master/packages/remark-parse#extending-the-parser for more on creating remark plugins